controllers/limitador_controller_ports_test.go

```diff
@@ -0,0 +1,323 @@
+			// Deployment command line
+			// Deployment probes
+			// Limitador CR status
+			// Service
+
+			deployment := appsv1.Deployment{}
+			Eventually(func() bool {
+				err := k8sClient.Get(
+					context.TODO(),
+					types.NamespacedName{
+						Namespace: testNamespace,
+						Name:      limitador.DeploymentName(limitadorObj),
+					},
+					&deployment)
+
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+
+			Expect(deployment.Spec.Template.Spec.Containers).To(HaveLen(1))
+			Expect(deployment.Spec.Template.Spec.Containers[0].Ports).To(ContainElements(
+				v1.ContainerPort{
+					Name: "http", ContainerPort: httpPortNumber, Protocol: v1.ProtocolTCP,
+				},
+				v1.ContainerPort{
+					Name: "grpc", ContainerPort: grpcPortNumber, Protocol: v1.ProtocolTCP,
+				},
+			))
+
+			Expect(deployment.Spec.Template.Spec.Containers[0].Command).To(
+				HaveExactElements(
+					"limitador-server",
+					"--http-port",
+					strconv.Itoa(int(httpPortNumber)),
+					"--rls-port",
+					strconv.Itoa(int(grpcPortNumber)),
+					"/home/limitador/etc/limitador-config.yaml",
+					"memory",
+				),
+			)
+
+			Expect(deployment.Spec.Template.Spec.Containers[0].LivenessProbe).NotTo(BeNil())
+			Expect(deployment.Spec.Template.Spec.Containers[0].LivenessProbe.
+				ProbeHandler.HTTPGet).NotTo(BeNil())
+			Expect(deployment.Spec.Template.Spec.Containers[0].LivenessProbe.
+				ProbeHandler.HTTPGet.Port).To(Equal(intstr.FromInt(int(httpPortNumber))))
+
+			Expect(deployment.Spec.Template.Spec.Containers[0].ReadinessProbe).NotTo(BeNil())
+			Expect(deployment.Spec.Template.Spec.Containers[0].ReadinessProbe.
+				ProbeHandler.HTTPGet).NotTo(BeNil())
+			Expect(deployment.Spec.Template.Spec.Containers[0].ReadinessProbe.
+				ProbeHandler.HTTPGet.Port).To(Equal(intstr.FromInt(int(httpPortNumber))))
+
+			limitadorCR := &limitadorv1alpha1.Limitador{}
+			err := k8sClient.Get(context.TODO(), client.ObjectKeyFromObject(limitadorObj), limitadorCR)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(limitadorCR.Status.Service).NotTo(BeNil())
+			Expect(limitadorCR.Status.Service.Ports).To(Equal(
+				limitadorv1alpha1.Ports{GRPC: grpcPortNumber, HTTP: httpPortNumber},
+			))
+
+			service := &v1.Service{}
+			err = k8sClient.Get(context.TODO(), types.NamespacedName{
+				Namespace: testNamespace,
+				Name:      limitador.ServiceName(limitadorObj),
+			}, service)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(service.Spec.Ports).To(ContainElements(
+				v1.ServicePort{
+					Name: "http", Port: httpPortNumber, Protocol: v1.ProtocolTCP,
+					TargetPort: intstr.FromString("http"),
+				},
+				v1.ServicePort{
+					Name: "grpc", Port: grpcPortNumber, Protocol: v1.ProtocolTCP,
+					TargetPort: intstr.FromString("grpc"),
+				},
+			))
+		})
+	})
+
+	Context("Updating limitador object with new custom ports", func() {
+		var limitadorObj *limitadorv1alpha1.Limitador
+
+		var httpPortNumber int32 = 8000
+		var grpcPortNumber int32 = 8001
```

Comment:
One trick that used in the past is to apply an offset to the default ports e.g `limitadorv1alpha1.DefaultServiceHTTPPort + 100` … but not asking to change anything. Just sharing.